dvc/external_repo.py

```diff
@@ -62,16 +62,33 @@ def _external_repo(url=None, rev=None, cache_dir=None):
 
     # Adjust new clone/copy to fit rev and cache_dir
     repo = Repo(new_path)
+    original_repo = None
     try:
         if rev is not None:
             repo.scm.checkout(rev)
+        # check if the URL is local and no default remote
+        # add default remote pointing to the original repo's cache location
+        if not os.path.isdir(url):
+            # Adjust original repo for pointing remote towards its' cache
+            original_repo = Repo(url)
+            rconfig = RemoteConfig(original_repo.config)
+            if not _remote_config_exists(rconfig):
+                rconfig.add(
+                    "upstream",
+                    original_repo.cache.local.cache_dir,
+                    default=True,
+                )
+                original_repo.scm.add([original_repo.config.config_file])
+                original_repo.scm.commit("add remote")
```

Comment:
This doesn't work. You already have a clone of the original repo, and now you are modifying the original repo, while not updating the clone. You need to adjust the clone and you can do it without committing, see comment above and see how we do that with `cache_dir` down below.